chainer/reporter.py

```diff
@@ -0,0 +1,325 @@
+import collections
+import contextlib
+
+import numpy
+import six
+
+from chainer import cuda
+from chainer import variable
+
+
+class Reporter(object):
+
+    """Object to which observed values are reported.
+
+    Reporter is used to collect values that users want to watch. The reporter
+    object holds a mapping from value names to the actually observed values.
+
+    When a value is passed to the reporter, an observer object can be
+    optionally attached. In this case, the name of the observer is added as the
+    prefix of the value name. The observer name should be registered
+    beforehand.
+
+    See the following example::
+
+       >>> from chainer import Reporter, report, report_scope
+       >>>
+       >>> reporter = Reporter()
+       >>> observer = object()  # it can be an arbitrary (reference) object
+       >>> reporter.add_observer('my_observer:', observer)
+       >>> observation = {}
+       >>> with reporter.scope(observation):
```

Comment:
In this example, `self.observation` is temporarily substituted to `observation` object by `scope` method. This behavior is not explained in the above description. So we should add the explanation that `scope` substitutes the observation to which observed values are stored.